src/newgrf_engine.cpp

```diff
@@ -458,30 +451,22 @@ static uint32_t VehicleGetVariable(Vehicle *v, const VehicleScopeResolver *objec
 				}
 
 				/* Pick the most common cargo type */
-				uint common_cargo_best_amount = 0;
-				for (CargoID cargo = 0; cargo < NUM_CARGO; cargo++) {
-					if (common_cargoes[cargo] > common_cargo_best_amount) {
-						common_cargo_best_amount = common_cargoes[cargo];
-						common_cargo_type = cargo;
-					}
-				}
+				auto cargo_it = std::max_element(std::begin(common_cargoes), std::end(common_cargoes));
+				/* Return CT_INVALID if nothing is carried */
+				CargoID common_cargo_type = (*cargo_it == 0) ? (CargoID)CT_INVALID : static_cast<CargoID>(cargo_it - std::begin(common_cargoes));
```

Comment:
Wouldn't `std::distance(std::begin(common_cargoes), cargo_it)` be conceptually better than `cargo_it - std::begin(common_cargoes)`? After all, if you're using `std::begin(common_cargoes)` over `common_cargoes.begin()`, then you're essentially saying that `common_cargoes` is some type with an iterator, so can you assume it is a contiguous block of memory?
Not sure whether a coding style / guideline / consensus about `std::begin(x)` vs `x.begin()` has been made, though `std::distance(x, y)` vs `y - x` should probably result in a similar result.

Reply:
`std::distance()` works for me. it shows intent.